### ISSUE TITLE
Please clarify that floats are not supported and do not attempt to parse them

### ISSUE THREAD

**Issue by tliron** [primetype/cbor_event#16]
Right now, you would only find this out in the Crates documentation. And, honestly, it is a fairly tiny note for what is ostensibly a major missing feature.

Otherwise, the library does attempt to parse floats, except we get wrong data. It seems that the developer started to work on an implementation but didn't finish it. I think it would be better to return an error rather than wrong data.

**Comment by NicolasDP** [comment]
You are right. Floating points are not supported (yet?) and this is not well enough documented.

**Comment by tliron** [comment]
For what it's worth, I switched to [Borc](https://codeberg.org/nasado/borc). Perhaps you can borrow some solutions from Borc? Hopefully they can work for no-std. (There are a million CBOR libraries for Rust, but almost none that are event-based and comprehensive.)

Thank you for your library!

**Comment by NicolasDP** [comment]
Thanks for letting me know. `Borc` has an interesting approach and it looks like they took seriously to be close to the RFC. I do not think they are `no_std` though. But please let me know if this is the case. I know @SimonIT is interested about a `no_std` CBOR library too.

`cbor_event` is an old project I keep wanting to come back to in order to address some API issues (not enough event based for example). Also, I'd like to remove the `Serialise` and `Deserialise` traits (maybe put it in a different crate?) as they don't really make sense in the context of event based API. Hopefully I should have some time to work on that in a few weeks and also remove the dependency on `std`.

**Comment by SimonIT** [comment]
I'm not sure what you mean by getting wrong data when parsing floats. We used `f32` successfully in a project with the current implementation (tho with [cddl-codegen](https://github.com/dcSpark/cddl-codegen) and my open PR there). `f64` should also work, there are tests for it:

https://github.com/primetype/cbor_event/blob/2b6330abe7aa69d230454ea2cadda5015526e681/src/de.rs#L957-L975

https://github.com/primetype/cbor_event/blob/2b6330abe7aa69d230454ea2cadda5015526e681/src/se.rs#L911-L933

We also used [minicbor](https://github.com/twittner/minicbor) as an alternative for true `no_std` support, because #15 was kinda hard to make

**Comment by tliron** [comment]
> I'm not sure what you mean by getting wrong data when parsing floats.

I mean that I used cbor_event to parse in some data and the floats all had wrong values in `Special::Float`. Integers and other values were fine. When that happened and I tried to look into it, I discovered the note in [the documentation](https://docs.rs/cbor_event/2.4.0/cbor_event/) that specials are supported "except floating points".

**Comment by NicolasDP** [comment]
My bad, it was done here #10

**Comment by tliron** [comment]
Yay! I have moved to another library (borc) as I don't have a need for no_std, but I'm sure this library will be very valuable to some users.

**Comment by SimonIT** [comment]
> My bad, it was done here #10

Yes, that's why I'm a bit confused, where did the wrong values come from? Maybe you were using `f16`? This is not supported as rust had no `f16` type, but as far as I can see at the moment they are planning something like this: rust-lang/rust#116909

**Comment by SebastienGllmt** [comment]
I updated the docs to mention the current state: #28 

The last thing would be that floats (unlike other types) don't support round-trip serialization (I also added this in the docs). This issue can be kept open to track this remaining point (although the issue name is somewhat stale), or it can be closed and a new issue can be opened to track that
